scripts/generate_conq_hose_manipulation_dataset.py

```diff
@@ -163,14 +163,15 @@ def main():
 
     now = int(time.time())
     root = Path(f"data/conq_hose_manipulation_data_{now}")
-    recorder = ConqDataRecorder(root, robot_state_client, image_client)
 
     with (LeaseKeepAlive(lease_client, must_acquire=True, return_at_exit=True)):
         command_client = setup_and_stand(robot)
 
         clients = Clients(lease=lease_client, state=robot_state_client, manipulation=manipulation_api_client,
                           image=image_client, raycast=rc_client, command=command_client, robot=robot,
-                          recorder=recorder)
+                          recorder=None)
+        recorder = ConqDataRecorder(root, clients=clients)
+        clients.recorder = recorder
```

Comment:
This circular reference/instantiation is giving me a bit of a pause. Though, I'm not sure if this circular reference is due to the demo structure or if it's due to the structure of the classes. I've got a few questions:
- Does it make sense for the `Clients` dataclass to hold a reference to the `recorder` or should these be separate?
- If `Clients` should have access to the `recorder`, does it make sense to instead set the `Clients.recorder` in the `ConqDataRecorder` constructor (likely doing some error-checking and raising an exception if `recorder` is already set)?

Reply:
Yea, we should probably remove the `recorder` member of the `clients` object, I don't think it's used anywhere anywas. I had it in there, but it probably shouldn't be.

Reply:
Removed the recorder from `clients` and adjusted the script c09fee108d71d61dc29362c651aaaf751b25bc5d